### PR TITLE
Use links to edition guide for edition migrations

### PR DESCRIPTION
**Pull request by ehuss**
This switches the migration lints for the 2024 edition to point to the edition guide documentation instead of the tracking issues. I expect the documentation should be easier to understand for a user, compared to most of the issues which don't have any direct information, and can be a bit confusing to navigate, or have outdated information.

### CODE REVIEW

**Comment by jieyouxu** [comment]
r? jieyouxu

File: compiler/rustc_lint/src/builtin.rs [hunk 1]
```diff
@@ -1814,7 +1814,7 @@ declare_lint! {
     "detects edition keywords being used as an identifier",
     @future_incompatible = FutureIncompatibleInfo {
         reason: FutureIncompatibilityReason::EditionError(Edition::Edition2024),
-        reference: "issue #49716 <https://github.com/rust-lang/rust/issues/49716>",
+        reference: "<https://doc.rust-lang.org/nightly/edition-guide/rust-2024/gen-keyword.html>",
```
Comment (jieyouxu) [inline compiler/rustc_lint/src/builtin.rs:1817]: Question (in general): we would have to change this to a non-nightly path once 2024 edition guide is stabilized right?

**Review by jieyouxu — APPROVED** [review]
Thanks!

File: compiler/rustc_lint/src/builtin.rs (hunk 1 above)
Comment (ehuss) [reply compiler/rustc_lint/src/builtin.rs:1817]: Not really, no. We use `/nightly/` URLs all over the compiler. The edition guide in particular is somewhat stable, in that it is unlikely `/nightly/` will introduce something that is *wrong* for a stable user. Edition changes are typically (though not always) fixed once it launches. There's also the benefit that users will see the most up-to-date docs with any fixes or clarifications.

That being said, I would think that `/stable/` URLs would also be fine once on the stable channel, but the compiler doesn't really have that capability. In cargo, we use an [automated linking system](https://github.com/rust-lang/cargo/blob/addcc8ca715bc7fe20df66afd6efbf3c77ef43f8/src/cargo/core/features.rs#L1405-L1414) to choose the channel. But I don't think that would work very well with compiletest validating the output.

Comment (jieyouxu) [reply compiler/rustc_lint/src/builtin.rs:1817]: Right, that is fine, thanks for the context!

**Pull request merged**